src/autowiring/DispatchQueue.cpp

```diff
@@ -87,12 +54,82 @@ void DispatchQueue::DispatchEventUnsafe(std::unique_lock<std::mutex>& lk) {
   (*thunk)();
 }
 
+void DispatchQueue::Abort(void) {
+  std::lock_guard<std::mutex> lk(m_dispatchLock);
+  m_aborted = true;
+  
+  // Do not permit any more lambdas to be pended to our queue:
+  m_dispatchCap = 0;
+  
+  // Destroy the whole dispatch queue:
+  while(!m_dispatchQueue.empty()) {
+    delete m_dispatchQueue.front();
+    m_dispatchQueue.pop_front();
+  }
+  
+  // Wake up anyone who is still waiting:
+  m_queueUpdated.notify_all();
+}
+
 bool DispatchQueue::DispatchEvent(void) {
   std::unique_lock<std::mutex> lk(m_dispatchLock);
-  if(m_dispatchQueue.empty())
+
+  // Update queue with delayed events that are ready
+  PromoteReadyEventsUnsafe();
+
+  if (m_dispatchQueue.empty())
```

Comment:
Recommend a change to this:

    if (m_dispatchQueue.empty() && PromoteReadyEventsUnsafe())